media-gfx/freecad/freecad-0.16.ebuild

```diff
@@ -0,0 +1,143 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 )
+
+inherit cmake-utils eutils fortran-2 multilib python-single-r1 fdo-mime
+
+DESCRIPTION="QT based Computer Aided Design application"
+HOMEPAGE="http://www.freecadweb.org/"
+SRC_URI="https://github.com/FreeCAD/FreeCAD/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+COMMON_DEPEND="dev-cpp/eigen:3
+	dev-libs/boost
+	dev-libs/xerces-c[icu]
+	dev-python/matplotlib
+	dev-python/pyside[X]
+	dev-python/shiboken
+	dev-qt/designer:4
+	dev-qt/qtgui:4
+	dev-qt/qtopengl:4
+	dev-qt/qtsvg:4
+	dev-qt/qtwebkit:4
+	media-libs/coin
+	|| ( sci-libs/opencascade:6.9.0[vtk] sci-libs/opencascade:6.8.0 sci-libs/opencascade:6.7.1 )
+	sys-libs/zlib
+	virtual/glu
+	${PYTHON_DEPS}"
+RDEPEND="${COMMON_DEPEND}
+	dev-qt/assistant:4
+	dev-python/pivy
+	dev-python/numpy"
+DEPEND="${COMMON_DEPEND}
+	dev-python/pyside-tools
+	>=dev-lang/swig-2.0.4-r1:0"
+
+# https://bugs.gentoo.org/show_bug.cgi?id=352435
+# https://www.gentoo.org/foundation/en/minutes/2011/20110220_trustees.meeting_log.txt
+RESTRICT="mirror"
+
+# TODO:
+#   DEPEND and RDEPEND:
+#		salome-smesh - science overlay
+#		zipio++ - not in portage yet
+
+S="${WORKDIR}/FreeCAD-${PV}"
+
+DOCS=(README.md ChangeLog.txt)
+
+pkg_setup() {
+	fortran-2_pkg_setup
+	python-single-r1_pkg_setup
+
+	[ -z "${CASROOT}" ] && die "empty \$CASROOT, run eselect opencascade set or define otherwise"
+}
+
+src_prepare() {
+	epatch "${FILESDIR}"/${PN}-0.14.3702-install-paths.patch
```

Comment:
eschew `epatch` in EAPI 6, stick it in `PATCHES`, call 1. `default` 2. the `sed` line and remove `eapply_user`